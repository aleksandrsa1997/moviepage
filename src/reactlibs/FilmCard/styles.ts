import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { Tooltip, withStyles} from '@material-ui/core';

export const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: '85%',
            height: '90%',
            margin: '15px',
            padding: theme.spacing(1),
        },
        media: {
            height: 320,
            objectFit : 'contain',
            margin: '7px',
        },
        fab: {
            height: '30px',
            width: '30px',
            margin: theme.spacing(0),
        },
    }),
);

export const HtmlTooltip = withStyles((theme: Theme) => ({
    tooltip: {
        backgroundColor: "#f5f5f9",
        color: "rgba(0, 0, 0, 0.87)",
        maxWidth: 220,
        fontSize: theme.typography.pxToRem(12),
        border: "1px solid #dadde9",
    }
}))(Tooltip);
