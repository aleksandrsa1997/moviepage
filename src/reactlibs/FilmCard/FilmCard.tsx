import React from 'react';
import Card from '@material-ui/core/Card';
import Rating from '@material-ui/lab/Rating';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import CardContent from '@material-ui/core/CardContent';
import CardActionArea from '@material-ui/core/CardActionArea';
import Tooltip from '@material-ui/core/Tooltip';
import { useStyles } from './styles';
import AddIcon from '@material-ui/icons/Add';
import Fab from '@material-ui/core/Fab';

export type TProps = {
    id: number,
    alt?: string,
    title?: string,
    color?: 'initial' | 'inherit' | 'primary' | 'secondary' | 'textPrimary' | 'textSecondary' | 'error',
    width?: number,
    image?: string,
    genre?: string,
    height?: number,
    onClick?:  (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
    variant?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'subtitle1' | 'subtitle2' | 'body1' | 'body2' | 'caption' | 'button' | 'overline' | 'srOnly' | 'inherit',
    overview?: string,
    textColor?: string,
    vote_count?: number,
    popularity?: number,
    vote_average?: number,
    original_title?: string,
};

const testImage = 'https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/best-kids-movies-2020-mulan-1571175006.jpg';

const FilmCard = ({
       id = 1,
       alt = 'Movie Poster',
       title = 'Raya and the Last Dragon',
       image = testImage,
       variant = 'h6',
       vote_count = 1,
       popularity = 1.5,
       vote_average = 8.5,
       overview = '',
       original_title = '',
       onClick,
 }: TProps) => {

    const classes = useStyles();

    return (
        <Card className={classes.root}>
            <CardActionArea onClick={onClick}>
                <Tooltip title={
                            <>
                                <Typography color="inherit">{`Original title: ${original_title}`}</Typography>
                                <em>{overview}</em> <br/> <b>{`Vote Average: ${vote_average}`}</b> <br/> <b>{`Vote Count: ${vote_count}`}</b>.{' '}
                            </>
                         }
                         aria-label='add'>
                    <Fab color="primary" className={classes.fab}>
                        <AddIcon />
                    </Fab>
                </Tooltip>
                <CardMedia
                    className={classes.media}
                    component="img"
                    alt={alt}
                    image={image}
                    id={String(id)}
                />
                <CardContent>
                    <Typography gutterBottom variant='h5'>
                        {title}
                    </Typography>
                    <Rating size={'medium'} defaultValue={vote_average} max={10} precision={0.5} readOnly/>
                    <Typography gutterBottom variant={variant}>
                        {'Vote count: ' + `${vote_count}`}
                    </Typography>
                    <Typography gutterBottom variant={variant}>
                        {'Popularity: ' + `${popularity}`}
                    </Typography>
                </CardContent>
            </CardActionArea>
        </Card>
    );
}

export default FilmCard;