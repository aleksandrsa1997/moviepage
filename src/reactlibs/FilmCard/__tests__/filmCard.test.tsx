import React from 'react';
import { act } from 'react-dom/test-utils';
import FilmCard, { TProps } from '../FilmCard';
import {mount, ReactWrapper} from 'enzyme';

jest.mock('react-i18next', () => ({
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

describe('FilmCard tests', () => {
    let props: TProps;
    let component: ReactWrapper<any, React.Component["state"], React.Component>;

    beforeEach(()=> {
        props = {
            alt: 'Movie Poster',
            title: 'Raya and the Last Dragon',
            image: 'https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/best-kids-movies-2020-mulan-1571175006.jpg',
            genre: 'comedy',
            variant: 'h6',
            vote_count: 1,
            popularity: 1.5,
            vote_average: 8.5,
            onClick: jest.fn()
        }
        component = mount(<FilmCard {...props}/>);
    })

    it('should render snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should call onClick function', () => {
        act(() => {
            component.find('FilmCard').at(0).getElement().props.onClick();
        })
        component.update();
        expect(props.onClick).toHaveBeenCalled();
    })
});