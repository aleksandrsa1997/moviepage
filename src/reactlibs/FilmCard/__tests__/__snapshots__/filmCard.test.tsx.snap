// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard tests should render snapshot 1`] = `ReactWrapper {}`;
